design-proposals/dynamic-primary-pod-nic-name.md

```diff
@@ -0,0 +1,298 @@
+Support Dynamic Primary Pod NIC Name
+=
+
+# Summary
+
+This proposal proposes a mechanism to dynamically determine the primary pod interface for KuveVirt virtual machines (
+VMs).
+The goal is to move away from the hard coded `eth0` interface and allow flexible, environment-driven interface
+selection.
+
+# Overview
+
+## Motivation
+
+In KubeVirt, each VM runs as a Kubernetes pod, and these pods have network interfaces attached to different networks.
+The default primary pod interface is often named `eth0`.
+However, users might require custom network setups where the pod’s primary interface is not named `eth0`.
+Currently, KubeVirt attaches the VM’s interface destined to the pod network to the pod’s `eth0` interface.
+
+When creating a VM with secondary networks, KubeVirt specifies the pod interfaces’ names for all secondary interfaces.
+It is currently not possible to specify the primary interface’s name when templating the virt-launcher pod, since there
+is no K8s API that enables it.
+
+## Goals
+
+- Allow the primary pod interface name to be dynamically determined based on the environment.
+- Ensure backward compatibility, so existing VMs with `eth0` as the primary interface do not break.
+- Handle the scenario of a non-eth0 primary interface with a binding plugin.
+
+## Non Goals
+
+- Change Kubernetes' API.
+- Change Multus' API.
+
+## Definition of Users
+
+- Cluster Administrator
+- VM owner
+
+## User Stories
+
+- As a cluster administrator I want to use a CNI that configures pods’ primary interface to a name other than `eth0`.
+
+- As a VM owner I want my VMs to be connected to the cluster's default network, as they do today.
+
+- As a VM owner I want a correct interface status reporting.
+
+## Repos
+
+- https://github.com/kubevirt/kubevirt
+
+# Design
+
+## Using Multus’ network-status annotation
+
+When KubeVirt is used in a cluster that has Multus as its default CNI, Multus adds the
+`k8s.v1.cni.cncf.io/network-status` annotation
```

Comment:
Since KubeVirt could be used with an arbitrary CNI, I don't think it should look for CNI-specific annotations.
Having said that, Multus is an exception since KubeVirt is currently coupled to it.